### PR TITLE
tests: kernel: Add description and group tests for doxygen

### PR DESCRIPTION
**Pull request by spoorthik**
Updated test description for tickless test cases, grouped test cases
as per doxygen requirements and some uncrustify's cosmetic changes.

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

File: tests/kernel/tickless/tickless/src/timestamps.c [hunk 1]
```diff
@@ -70,16 +70,16 @@ void _timestamp_open(void)
...
-					   */
-	_TIMESTAMP_IMASK = 0x70F;  /* mask all timer interrupts */
-	_TIMESTAMP_ICLEAR = 0x70F;  /* clear all interrupt status */
-
-	_TIMESTAMP_CTRL = 0x1;  /* enable timer */
+	_TIMESTAMP_CTRL = 0x0;                  /* disable/reset timer */
+	_TIMESTAMP_CFG = 0x0;                   /* 32-bit timer */
+	_TIMESTAMP_MODE = 0x2;                  /* periodic mode */
+	_TIMESTAMP_LOAD = _TIMESTAMP_MAX;       /* maximum interval
+											 * to reduce rollovers
```
Comment (nashif) [inline tests/kernel/tickless/tickless/src/timestamps.c]: fix those comments please.

File: tests/kernel/tickless/tickless/src/timestamps.c [hunk 2]
```diff
@@ -175,22 +175,24 @@ void _timestamp_open(void)
...
+	_TIMESTAMP_RACCESS = 0xFF;      /* allow register read access */
+	_TIMESTAMP_WACCESS = 0xFF;      /* allow register write access */
+	_TIMESTAMP_IMASK = 0x0;         /* mask all timer interrupts */
 
 	/* minimum 0.3 sec delay required for oscillator stabilization */
 	k_sleep(0.3 * MSEC_PER_SEC);
 
-	_TIMESTAMP_VAL = 0x0;  /* clear invalid time flag in status register */
+	_TIMESTAMP_VAL = 0x0;           /* clear invalid time flag in
+									 * status register
```
Comment (nashif) [inline tests/kernel/tickless/tickless/src/timestamps.c]: fix those comments please.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/tickless/tickless/src/main.c [hunk 3]
```diff
@@ -199,13 +199,26 @@ void ticklessTestThread(void)
...
+		      "* TEST FAILED. TICK COUNT INCORRECT *");
 
 	/* release the timer, if necessary */
 	_TIMESTAMP_CLOSE();
 
 }
+/**
+ * @brief Tests to verify tickless feature
+ *
+ * @defgroup kernel_tickless Tickless
```
Comment (nashif) [inline tests/kernel/tickless/tickless/src/main.c]: `@defgroup kernel_tickless_tests Tickless`

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by spoorthik** [comment]
@nashif : Addressed review comments. Please have a look.

File: tests/kernel/tickless/tickless_concept/src/main.c [hunk 4]
```diff
@@ -63,8 +63,17 @@ static void thread_tslice(void *p1, void *p2, void *p3)
 #endif
 	k_sem_give(&sema);
 }
+/**
+ * @addtogroup kernel_tickless
```
Comment (nashif) [inline tests/kernel/tickless/tickless_concept/src/main.c]: needs to be kernel_tickless_tests

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**